Fix mobile display for new email forward page

daledupreez opened the pull request:
#### Changes proposed in this Pull Request

* This PR ensures that the new email forward page is mobile-friendly
* It is worth noting that the implementation here shows the warning and button vertically instead of horizontally as per the design, as some translated strings would not work well horizontally.

#### Testing instructions

* Ensure that you have a domain with active email forwards, including at least one where the email address still needs to be verified.
* Open a local version of Calypso running this branch, or the live version.
* Navigate to `/email/:yourSiteSlug?flags=email/centralized-home` to bring up the new email management home page.
* Click through to your domain with email forwards
* Adjust the size of the display to test various mobile, tablet, and larger screen sizes.
* Verify that all the text, the verification warning, and the resend email verification button are always visible and clickable.
* Either switch languages to Russian/German (or some language of your choice), or manually make the text in the resend verification very long, and verify that the text is visible at various sizes. (The longest text for the button is for Russian: `Повторно отправить письмо для подтверждения`.)

#### Screenshot

<img width="383" alt="Screenshot 2021-05-12 at 15 44 39" src="https://user-images.githubusercontent.com/3376401/117985500-0b5bb780-b339-11eb-906b-9ccd8c459847.png">


Related to #52169, where this was originally introduced.

daledupreez committed:
First pass for strictly mobile layouts

daledupreez committed:
Fix warning font size and improve small tablet handling

daledupreez commented:
@saygunnyc, could you take a quick look here? I adjusted the layout to stay vertical in mobile/narrow layouts to cleanly support (much) longer translated text. For a visual Russian has the following width (I think I tweaked the vertical space after taking this shot):

<img width="417" alt="Screenshot 2021-05-11 at 13 42 45" src="https://user-images.githubusercontent.com/3376401/117814243-56f06180-b264-11eb-8482-45e78ef4e93d.png">

saygunnyc commented:
Thank you for the ping @daledupreez and the screenshots! Overall, looking great! 
A couple of notes below to consider, also sharing design here for reference.

<img width="372" alt="Screen Shot 2021-05-11 at 10 17 19 AM" src="https://user-images.githubusercontent.com/78876323/117831067-3dd6b900-b242-11eb-9693-e288537a57d4.png">


- I think the fixed-sized long CTA looks good in a case where we have a longer translated copy. It would be better if we still left-align the status message and (!) though. 
-  Can we have the vertical red line indicator as an overlay, so the email icon and ">" are still aligned the same as others and not indented? Right now they don't look aligned.
- I had another indicator in design to show how many email forwardings a user can have and how many they are using currently "3 of 5". Are we having issues with that approach?
- I haven't yet had the chance to reply to the comment on the P2 but, I think seeing "Mailboxes" as a title here confuses me as a user, as these are not mailboxes. And then what will we use for the CTA below? Add a new mailbox? I think it's also worth thinking about what we'll be using once we upsell an email plan here.
- Also, if there's an action required by the user, the intention was that the email product reflects that, and its status changes from "Active" to "Action Required". So if this is a long list of mailboxes, the user is aware at first sight that something is off with their email forwardings.

Lemme know your thoughts, and if these make sense...
Looking forward to seeing these rolled out!

daledupreez commented:
Thanks for the quick response, @saygunnyc!

I have a few specific responses, and then some questions.

> Also, if there's an action required by the user, the intention was that the email product reflects that, and its status changes from "Active" to "Action Required". So if this is a long list of mailboxes, the user is aware at first sight that something is off with their email forwardings.

The various error/warning states across items is something we're working on this week - we should be wrapping up the API improvements to support this today or tomorrow.

> * I haven't yet had the chance to reply to the comment on the P2 but, I think seeing "Mailboxes" as a title here confuses me as a user, as these are not mailboxes. And then what will we use for the CTA below? Add a new mailbox? I think it's also worth thinking about what we'll be using once we upsell an email plan here.
> * I had another indicator in design to show how many email forwardings a user can have and how many they are using currently "3 of 5". Are we having issues with that approach?

I missed both of these items in my previous PRs -- I'll tackle them in a follow-up PR, as this PR is specifically aimed at fixing the mobile layout.

> Can we have the vertical red line indicator as an overlay, so the email icon and ">" are still aligned the same as others and not indented? Right now they don't look aligned.

I suspect we're not adjusting a margin/padding here -- I'll fix this one.

#### Question

> I think the fixed-sized long CTA looks good in a case where we have a longer translated copy. It would be better if we still left-align the status message and (!) though.

I am not sure what you mean by this. Do we want a vertical layout for the warning and button, but keep the warning left-aligned? Or do we want a horizontal layout where they both get very tall when the text gets long?

saygunnyc commented:
Thanks so much, @daledupreez for clarifications! 

>I am not sure what you mean by this. Do we want a vertical layout for the warning and button, but keep the warning left-aligned? Or do we want a horizontal layout where they both get very tall when the text gets long?

Sharing a quick mockup below, hope this is helpful!

![quick_mock](https://user-images.githubusercontent.com/78876323/117843037-7a0f1700-b24c-11eb-953d-6de5e42f1f31.jpg)

Not sure if that's also WIP, but just noticed that the color of the vertical line is somewhat washed out, not RED 50.

daledupreez committed:
Left-align warning text in all views, not just desktop

daledupreez committed:
Fix alignment and colour for error card highlight

daledupreez commented:
I've addressed the indent for the error/warning state, as well as the colour for the warning state. I have also fixed the alignment for the warning, but kept the warning and related CTA vertical for narrow screens.

saygunnyc commented:
Thank you, @daledupreez!
> but kept the warning and related CTA vertical for narrow screens.

Would u mind letting me know the reasoning behind this, so I understand better?

daledupreez commented:
Not a problem, @saygunnyc! I based the update on your mockup in [this comment](https://github.com/Automattic/wp-calypso/pull/52750#issuecomment-838690223) above, where the warning and the CTA appear vertically instead of horizontally. I did that because I thought's that what you had proposed, so I am glad I called it out! 😁

saygunnyc commented:
Sorry for the additional comment! Now I understand what you mean. All good!

stephanethomas reviewed (changes requested):
I've noticed some display issues with the button for certain widths:

![screenshot](https://user-images.githubusercontent.com/594356/117991599-7f4c8e80-b33e-11eb-859c-74984b5afb43.png)

We could also make sure that the warning and the button are vertically centered (you can check what I did in the `Email Comparison` page). Given the length of each line, I wonder if we shouldn't switch to the column view much sooner, probably as soon as the viewport reaches 960 pixels (or maybe we should have breakpoints where the warning and button are displayed alone on the same line).

By the way, the button seems too narrow in small viewports, at least in comparison with other pages:

Email Forwarding Settings | Email Comparison
------ | -----
![screenshot](https://user-images.githubusercontent.com/594356/117992862-84f6a400-b33f-11eb-8736-72f927f963fa.png) | ![screenshot](https://user-images.githubusercontent.com/594356/117993099-b8d1c980-b33f-11eb-8eaf-339d1285bb95.png)

saygunnyc commented:
> By the way, the button seems too narrow in small viewports, at least in comparison with other pages

Yup, this is because I've been using the compact button for all inline _– any CTA within a card_ CTAs. Same styling with the "Renew" button. I guess I'm for keeping it that way unless you feel strongly about using the larger button style.

Below are two quick mockups so we can compare: 

<img width="991" alt="Screen Shot 2021-05-12 at 10 55 43 AM" src="https://user-images.githubusercontent.com/78876323/117997075-ac824800-b310-11eb-9641-2b31dfd2e1b1.png">
<img width="986" alt="Screen Shot 2021-05-12 at 10 55 52 AM" src="https://user-images.githubusercontent.com/78876323/117997076-ac824800-b310-11eb-9d9b-958a4505dd58.png">

daledupreez committed:
Restructure button HTML to fix layout; shift breakpoints

stephanethomas committed:
Make sure warning and button are centered

This also simplifies some rules while making sure the warning is always stuck to the first column.

stephanethomas commented:
This looks good now. I've just added a commit to make sure that items are vertically centered:

![screenshot](https://user-images.githubusercontent.com/594356/118134552-57befa00-b402-11eb-8fca-7b0ff8cbeea7.png)

daledupreez commented:
@stephanethomas, we may need to remove [the change](https://github.com/Automattic/wp-calypso/pull/52750/commits/10d3765e01e6976d96251e4b9260ad8b9b2fa003) you referenced in [this comment](https://github.com/Automattic/wp-calypso/pull/52750#issuecomment-840571399). 

The design for the email forwarding page has those elements aligned to the top of the cards:
<img width="986" alt="Screenshot 2021-05-13 at 16 14 28" src="https://user-images.githubusercontent.com/3376401/118138585-cdc56000-b406-11eb-82ba-d327256ec1a7.png">

daledupreez committed:
Revert "Make sure warning and button are centered"

This reverts commit 10d3765e01e6976d96251e4b9260ad8b9b2fa003.
